Source/MQTTnet.Extensions.Rpc/Options/TopicGeneration/TopicGenerationContext.cs

```diff
@@ -3,23 +3,30 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Collections.Generic;
 using MQTTnet.Client;
 using MQTTnet.Protocol;
 
 namespace MQTTnet.Extensions.Rpc
 {
     public sealed class TopicGenerationContext
     {
-        public TopicGenerationContext(IMqttClient mqttClient, MqttRpcClientOptions options, string methodName, MqttQualityOfServiceLevel qualityOfServiceLevel)
+        static readonly IDictionary<string, object> EmptyParameters = new Dictionary<string, object>();
+
+        public TopicGenerationContext(IMqttClient mqttClient, MqttRpcClientOptions options, string methodName,
+            IDictionary<string, object> parameters, MqttQualityOfServiceLevel qualityOfServiceLevel)
         {
             MethodName = methodName ?? throw new ArgumentNullException(nameof(methodName));
+            Parameters = parameters ?? EmptyParameters;
```

Comment:
Please leave the parameter _null_ because otherwise the user can modify the instance of the "empty" dictionary because it is not read only (immutable).

Reply:
Thanks @chkr1011 for review.
Would it be ok if I changed type for parameters from `IDictionary` to `IReadOnlyDictionary` ?

Reply:
That would indeed circumvent the issue but I am afraid that maybe some users want to change these values and use it as a way to return data to the caller as well. So I would leave it null here. If you want to avoid checking for null all the time (what I can understand) you may pass an "Empty" immutable instance on your own.

Reply:
Made parameters nullable as suggested.